src/ansys/fluent/core/solver/flobject.py

```diff
@@ -372,17 +388,49 @@ class String(SettingsBase[str], Textual):
     _state_type = str
 
 
-class Filename(SettingsBase[str], Textual):
+class InputOutputFileBase:
+    def is_input(self):
+        if self.file_purpose() == "input":
+            return True
+        return False
```

Comment:
Convoluted logic: a single-line does this.

Reply:
Yes, updated. Thank you.